espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/models/ActionSequence.java

```diff
@@ -0,0 +1,51 @@
+package io.appium.espressoserver.lib.helpers.w3c.models;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException;
+import io.appium.espressoserver.lib.handlers.exceptions.NotYetImplementedException;
+
+import static io.appium.espressoserver.lib.helpers.w3c.models.W3CActions.processSourceActionSequence;
+
+/**
+ * Pull out the actions from the input sources and group them by 'ticks'
+ *
+ * Defined in 17.3 of spec 'extract an action sequence'
+ */
+public class ActionSequence implements Iterator<Tick> {
+
+    private List<Tick> actionsByTick;
+    private int tickCounter;
+
+    public ActionSequence(W3CActions w3CActions) throws InvalidArgumentException, NotYetImplementedException {
+        tickCounter = 0;
+        actionsByTick = new ArrayList<>();
+
+        // Transpose the actionsByTick within the input sources
```

Comment:
Looks like I'm too stupid to understand this block of code :(

Reply:
I'll add a more detailed comment to the class description.

It's supposed to group actions together so that each `tick` has a list of `action objects`.